### ISSUE TITLE
@export _node : Node is always null in editor and triggers configuration warnings even when the field is filled

### ISSUE THREAD

**Issue by LiamSmyth** [godotengine/godot#77363]
### Godot version

4.0.2 stable

### System information

Windows 10

### Issue description

When trying to use configuration warnings, `@export` nodes are always null in editor whether something is plugged in or not, which means that when checking editor configuration warnings the @exported value will always be null, and hence trigger the warning. In the simplest case, this will always throw the warning even if a node is plugged in in editor.

```python
@tool
extends Node2D

@export var _node: Node

func _get_configuration_warnings() -> PackedStringArray:
    var warnings: PackedStringArray = []
    if _node == null:
        warnings = ["_node is null!"]
    return warnings
```

### Steps to reproduce

Add the above script to a node, add a child node, select the child as _node in the editor. Observe that the warning is still thrown even after scene reload

### Minimal reproduction project

n/a

**Comment by dalexeev** [comment]
@ajreckof Perhaps your PR #76389 also fixes this?

**Comment by ajreckof** [comment]
This a subcase of #74141 so it will effectively be fixed by #76389

**Comment by KoBeWi** [comment]
Closing as duplicate of #74141
